target_snowflake/__init__.py

```diff
@@ -135,6 +171,8 @@ def persist_lines(config, lines, information_schema_cache=None) -> None:
             # Get schema for this record's stream
             stream = o['stream']
 
+            adjust_timestamps_in_record(o['record'], schemas[stream])
```

Comment:
do we expect any performance degradation by running `adjust_timestamp_in_record` on every message? Or this should not add significant overhead.

Reply:
From a time complexity perspective, it doesn't add any overhead because there are some other functions in the `if t == 'RECORD': ` block,  such as `float_to_decimal(o['record'])` which also seems to walk through the record, the 3rd party one `validators[stream].validate` needs to also validate the record against the schema...etc.

Reply:
So basically, the whole if block's time complexity is the slowest function in it, and  adjust_timestamps_in_record is not the one.

Reply:
Hope this answers your question.

Reply:
absolutely, thank you